templates/base/client/.rtag/App.tsx.hbs

```diff
@@ -28,95 +29,137 @@ function App() {
+                  onConnect={(stateId) =>
+                    client
+                      .connectExisting(token, stateId, setState)
+                      .then((conn) => setConnection(conn))
+                      .catch((e) => toast.error("Connection error: " + e.reason))
                   }
-                }}
-              />
-            }
-          />
-          <Route
-            path="/"
-            element={
-              <Home
-                onConnect={(request) =>
-                  client
-                    .connectNew(token, request, setState)
-                    .then((conn) => {
-                      setConnection(conn);
-                      navigate(`/state/${conn.stateId}`);
-                    })
-                    .catch((e) => toast.error("Connection error: " + e.reason))
-                }
-              />
-            }
-          />
-          <Route path="*" element={<div>Not Found</div>} />
-        </Routes>
+                  onDisconnect={() => {
+                    if (connection !== undefined) {
+                      connection.disconnect();
+                      setConnection(undefined);
+                      setState(undefined);
+                    }
+                  }}
+                />
+              }
+            />
+            <Route
+              path="/"
+              element={
+                <Home
+                  onConnect={(request) =>
+                    client
+                      .connectNew(token, request, setState)
+                      .then((conn) => {
+                        setConnection(conn);
+                        navigate(`/state/${conn.stateId}`);
+                      })
+                      .catch((e) => toast.error("Connection error: " + e.reason))
+                  }
+                />
+              }
+            />
+            <Route path="*" element={<div>Not Found</div>} />
+          </Routes>
+        )}
       </div>
     </>
   );
 }
 
+function NavBar({ token }: { token?: string }) {
+  const navigate = useNavigate();
+  const user = token !== undefined ? RtagClient.getUserFromToken(token) : undefined;
+  return (
+    <div className="bg-indigo-500">
+      <div className="flex items-center justify-between h-16 px-4 mx-auto max-w-7xl sm:px-6 lg:px-8">
+        <div className="flex items-center">
+          <div className="flex-shrink-0 cursor-pointer" onClick={() => navigate(`/`)}>
+            <img
+              className="w-10 h-10"
+              src="https://i.ibb.co/sQMmTTq/hathora-hammer-logo-light.png"
+              alt="hathora-hammer-logo"
+            />
+          </div>
+          <div className="flex items-center flex-shrink-0 cursor-pointer" onClick={() => navigate(`/`)}>
+            <span className="mt-1 ml-3 text-2xl font-semibold text-white uppercase font-display">{{appName}}</span>
+          </div>
+          {user && (
+            <div className="hidden md:block">
```

Comment:
so right now the username disappears on mobile -- is there any other option we have?

Reply:
this has been fixed